src/PackageManagement/AzureAutomationTools.PackageManagement.psm1

```diff
@@ -767,27 +921,60 @@ function DeployModules {
             
         foreach ($Module in $Modules) {
             $ContentLink = $Module.Package
-            $Params = @{Name = $Module.Name; }
+            $Params = @{Name = $Module.Name;}
             $Params += $CommonParameters
-       
-            #FIX failing - version property is coming back blank
-            if ($ExistingModules.Where( {$_.Name -eq $Module.Name -and $_.Version -eq $Module.Version})) {
-                Write-Output "Existing module up to date - skipping."
-            }
-            elseif ($ExistingModules.Where( {$_.Name -eq $Module.Name })) {           
-                Write-Output "Updating existing module ..."
-            
-                Set-AzureRmAutomationModule -ContentLinkUri $ContentLink  @Params
+            $AutomationModule = $null
             
-                Write-Output "Updating existing module - done."
+            try {
+                $ExistingModule = Get-AzureRmAutomationModule @Params
+            }
+            catch [Microsoft.Azure.Commands.Automation.Common.ResourceNotFoundException] {
+                Write-Verbose "Cannot find module '$($Module.Name)' in automation account '$AutomationAccountName'."
+                $ExistingModule = $null
+            }
+
+            if ($null -eq $ExistingModule) {
```

Comment:
I try to avoid using the -not operator when testing against null as it's falsey. Can change this if we all agree.